packages/core-data/src/resolvers.js

```diff
@@ -261,37 +261,41 @@ export const getEntityRecords =
 				} );
 			}
 
-			dispatch.receiveEntityRecords(
-				kind,
-				name,
-				records,
-				query,
-				false,
-				undefined,
-				meta
-			);
+			registry.batch( () => {
+				dispatch.receiveEntityRecords(
+					kind,
+					name,
+					records,
+					query,
+					false,
+					undefined,
+					meta
+				);
 
-			// When requesting all fields, the list of results can be used to
-			// resolve the `getEntityRecord` selector in addition to `getEntityRecords`.
-			// See https://github.com/WordPress/gutenberg/pull/26575
-			if ( ! query?._fields && ! query.context ) {
-				const key = entityConfig.key || DEFAULT_ENTITY_KEY;
-				const resolutionsArgs = records
-					.filter( ( record ) => record[ key ] )
-					.map( ( record ) => [ kind, name, record[ key ] ] );
+				// When requesting all fields, the list of results can be used to
+				// resolve the `getEntityRecord` selector in addition to `getEntityRecords`.
+				// See https://github.com/WordPress/gutenberg/pull/26575
+				if ( ! query?._fields && ! query.context ) {
+					const key = entityConfig.key || DEFAULT_ENTITY_KEY;
+					const resolutionsArgs = records
+						.filter( ( record ) => record[ key ] )
+						.map( ( record ) => [ kind, name, record[ key ] ] );
+
+					dispatch( {
+						type: 'START_RESOLUTIONS',
+						selectorName: 'getEntityRecord',
+						args: resolutionsArgs,
+					} );
+					dispatch( {
+						type: 'FINISH_RESOLUTIONS',
+						selectorName: 'getEntityRecord',
+						args: resolutionsArgs,
+					} );
```

Comment:
If the start and finish actions are batched, then there's no point dispatching `START_RESOLUTIONS`. The state will be overwritten by `FINISH_RESOLUTIONS`, and no listener will see the state between start and finish because listeners are not called during a batch.

If some listener is observing the resolution state and expects to see the full cycle, `undefined` -> `resolving` -> `finished`, it might get suprised that it won't ever see `resolving`. It will go directly to `finished`. Not sure how much breaking this is.

Reply:
I think it probably doesn't really make sense to call "start" in this case, we just want to make these as resolved to avoid requests.